packages/@aws-cdk/aws-stepfunctions-tasks/lib/dynamodb/call-dynamodb.ts

```diff
@@ -106,7 +106,7 @@ export class DynamoAttributeValue {
    * Sets an attribute of type Binary. For example:  "B": "dGhpcyB0ZXh0IGlzIGJhc2U2NC1lbmNvZGVk"
    */
   public withB(value: string) {
-    this.attributeValue.B = value;
+    this.attributeValue.B = Buffer.from(value, 'base64');
```

Comment:
This isn’t what we want. Looking at TaskInput and other bits now

Reply:
It's looking like it's probably on the receiving end `arn:aws:states:::dynamodb:updateItem` as CDK is generating `"Parameters":{"Key":{"ContentID":{"B.$":"$.ContentID"}}` as input to that.

Reply:
Unless the `"'s` around $.ContentID in the above "Parameters" that are the issue?